central/sensor/service/connection/connection_impl.go

```diff
@@ -591,13 +591,27 @@ func (c *sensorConnection) Run(ctx context.Context, server central.SensorService
 			return errors.Wrapf(err, "unable to get audit log file state sync msg for %q", c.clusterID)
 		}
 
-		// Send the audit log state to sensor even if the the user has it disabled (that's set in dynamic config). When enabled, sensor will use it correctly
+		// Send the audit log state to sensor even if the user has it disabled (that's set in dynamic config). When enabled, sensor will use it correctly
 		if err := server.Send(msg); err != nil {
 			return errors.Wrapf(err, "unable to sync audit log file state to cluster %q", c.clusterID)
 		}
 	}
 
-	metrics.IncrementSensorConnect(c.clusterID, c.sensorHello.GetReconnect())
+	switch c.sensorHello.GetSensorState() {
+	case central.SensorHello_RECONNECT:
+		metrics.IncrementSensorConnect(c.clusterID, true, true)
+	case central.SensorHello_STARTUP:
+		metrics.IncrementSensorConnect(c.clusterID, false, true)
+	case central.SensorHello_UNKNOWN:
+		metrics.IncrementSensorConnect(c.clusterID, false, false)
+		log.Warnf("Unknown sensor state: %s. This means that sensor is running an older version than central and does not report its connection state",
+			c.sensorHello.GetSensorState())
+	default:
+		metrics.IncrementSensorConnect(c.clusterID, false, true)
```

Comment:
Unknown state should be `false, false`, right?

I think changing these bools to enum would help here.

Reply:
> Unknown state should be false, false, right?

The `default` state means that its a new state that Central does not know about yet. I.e. sensor is ahead of central and communicates this state. In this case, we can say that the feature *is* enabled, right?